src/test/java/org/folio/services/ftp/SftpUploadServiceTest.java

```diff
@@ -0,0 +1,116 @@
+  private static final Logger logger = LogManager.getLogger(SftpUploadServiceTest.class);
+  private static final String FILENAME = "batchVoucher.json";
+  private static final String USERNAME = "user";
+  private static final String PASSWORD = "password";
+  private static final String EXPORT_FOLDER_NAME = "upload";
+  private static final Context context = Vertx.vertx().getOrCreateContext();
+  private static String uri;
+  private static final String INVALID_URI = "ftp://localhost:11/";
+
+
+  @Container
+  public static final GenericContainer sftp = new GenericContainer(
+    new ImageFromDockerfile()
+      .withDockerfileFromBuilder(builder ->
+        builder
+          .from("atmoz/sftp:latest")
+          .run("mkdir -p " + File.separator + EXPORT_FOLDER_NAME + "; chmod -R 777 " + File.separator + EXPORT_FOLDER_NAME)
+          .build()))
+      .withExposedPorts(22)
+      .withCommand(USERNAME + ":" + PASSWORD + ":::upload");
+
+  @BeforeAll
+  public static void staticSetup() {
+    sftp.start();
+    uri = "ftp://"+sftp.getHost()+":"+sftp.getMappedPort(22)+"/";
+  }
+
+  @Test
+  public void testSuccessfulUpload(VertxTestContext vertxTestContext) throws Exception {
+    logger.info("=== Test successful upload ===");
+    Date end = new Date();
+    end.setTime(System.currentTimeMillis() - 864000000);
+
+    BatchVoucher batchVoucher = new BatchVoucher();
+    batchVoucher.setId(UUID.randomUUID().toString());
+    batchVoucher.setStart(end);
+    batchVoucher.setEnd(end);
+    batchVoucher.setBatchGroup(UUID.randomUUID().toString());
+    batchVoucher.setCreated(new Date());
+
+    SftpUploadService helper = new SftpUploadService(context, uri);
+    var future = helper.upload(USERNAME, PASSWORD, EXPORT_FOLDER_NAME, FILENAME , JsonObject.mapFrom(batchVoucher).encodePrettily())
+      .onSuccess(logger::info)
+      .onFailure(t -> {
+        logger.error(t);
+        Assertions.fail(t.getMessage());
+      })
+        .onComplete(logger::info);
+    vertxTestContext.assertComplete(future)
+      .onSuccess(result -> vertxTestContext.completeNow());
+  }
+
+  @Test
+  void testSuccessfulUploadForLongPath(VertxTestContext vertxTestContext) throws Exception {
+    logger.info("=== Test successful upload ===");
+    Date end = new Date();
+    end.setTime(System.currentTimeMillis() - 864000000);
+
+    BatchVoucher batchVoucher = new BatchVoucher();
+    batchVoucher.setId(UUID.randomUUID().toString());
+    batchVoucher.setStart(end);
+    batchVoucher.setEnd(end);
+    batchVoucher.setBatchGroup(UUID.randomUUID().toString());
+    batchVoucher.setCreated(new Date());
+
+    SftpUploadService helper = new SftpUploadService(context, uri);
+    var future = helper.upload(USERNAME, PASSWORD, EXPORT_FOLDER_NAME+"/test/long/path", FILENAME , JsonObject.mapFrom(batchVoucher).encodePrettily())
+      .onSuccess(logger::info)
+      .onFailure(t -> {
+        logger.error(t);
+        Assertions.fail(t.getMessage());
```

Comment:
Since we introduced test containers for ftp here, it would be good to use it capabilities and write full integration test to query existed file from ftp/sftp